config/config.go

```diff
@@ -369,13 +417,28 @@ The parents attribute is implicit from the parent declaration in the document`)
 		parsedDoc.Schema.Attributes[parsedName] = &parsedAttr
 	}
 
-	parsedDoc.ParentReqSpec.Level = doc.Parent.Level
-	parsedDoc.ParentReqSpec.Prefix = doc.Parent.Prefix
+	// Add the parents attribute and link specifications
 	if doc.Parent.Level != "" && doc.Parent.Prefix != "" {
-		// Add the parents attribute
+		link, err := parseParent(doc.Parent, doc.Prefix, doc.Level)
+		if err != nil {
+			return err
+		}
+		parsedDoc.LinkSpecs = append(parsedDoc.LinkSpecs, link)
+	}
+	if doc.Parents != nil {
+		for _, p := range doc.Parents {
+			link, err := parseParent(p, doc.Prefix, doc.Level)
+			if err != nil {
+				return err
+			}
+			parsedDoc.LinkSpecs = append(parsedDoc.LinkSpecs, link)
+		}
+
+	}
```

Comment:
As you mentioned, I think we could be somewhat polymorphic and use a single key and parse it differently depending on whether the input is an object or an array. I think that makes it easier than remembering that there is one variant with an `s` and another one without it.

But if we go for the 2 separate variants, maybe we should make them exclusive to flag the case where a user defines both, which doesn't make much sense.

Reply:
turned out the polymorphic route was simpler than expected